src/java.base/unix/native/libnio/ch/Net.c

```diff
@@ -571,6 +571,8 @@ Java_sun_nio_ch_Net_setIntOption0(JNIEnv *env, jclass clazz, jobject fdo,
     }
 }
 
+jint handleSocketErrorWithMessage(JNIEnv *env, jint errorValue, const char* message);
```

Comment:
This looks okay but we should probably put the function prototype at the top rather in the middle of the JNI functions. I guess it can be static too.